polkadot/runtime/rococo/src/lib.rs

```diff
@@ -428,18 +426,14 @@ impl pallet_treasury::Config for Runtime {
 	type WeightInfo = weights::pallet_treasury::WeightInfo<Runtime>;
 	type SpendFunds = Bounties;
 	type SpendOrigin = TreasurySpender;
-	type AssetKind = VersionedLocatableAsset;
+	type AssetKind = VersionedAssetId;
```

Comment:
I don't think this is actually enough; `LocatableAssetId` had both `Location` and an `Asset`, the former is where the latter is located, e.g. I can either have DOT tokens on the relay chain or on the Asset Hub, or any other system parachain, and the `Location` parameter is used to distinguish where the `Asset` is located.

Reply:
Yes, `AssetId` doesn't hold the same information as `LocatableAssetId`, it's just a simplification because I moved the other piece of information, the location where the asset is held, elsewhere.

The `PayOverXcm` trait always required the account to be in the actual destination chain, given that it did a `TransferAsset` to that account on the destination chain.
Given that, the information of where to send the message should be encoded in the location of the beneficiary itself, which is what I did.

So before we had this:
- beneficiary -> specifies the recipient account on the destination chain, but doesn't specify the destination chain
- asset kind -> specifies the asset and the location where it's held, which is used as the destination chain
This means there's a tight coupling between beneficiary and asset kind because the location to send the message shouldn't live alongside the asset.

Now we have:
- beneficiary -> specifies the destination chain and the recipient account within it
- asset -> just specifies the asset to transfer

I agree you can either have DOT on the relay or asset hub, but the asset is still the same, DOT, you just need to specify where you send the message. Every receiver interprets the asset as they need to to come up with DOT.